integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.esb.message.store.test;
+
+import org.apache.axiom.om.OMElement;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+
+import static org.testng.Assert.assertTrue;
+
+/**
+ * This test case tests whether a message can be routed dynamically by
+ * deriving the store name from the request.
+ */
+public class DynamicMessageRoutingTestCase extends ESBIntegrationTest {
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+    }
```

Comment:
It's better to verify the proxy service existence here by invoking,
verifyProxyServiceExistence("storeDynamicRoutingTestProxy");